core/domain/exp_services.py

```diff
@@ -431,6 +431,17 @@ def export_to_zip_file(exploration_id, version=None):
     return memfile.getvalue()
 
 
+def convert_state_dict_to_yaml(state_dict, width=80):
+    try:
+        # Check if the state_dict can be converted to a State.
+        state = exp_domain.State.from_dict(state_dict)
+    except Exception:
+        logging.info('Bad state dict: %s' % str(state_dict))
+        raise Exception('Could not convert state dict to YAML.')
+
+    return utils.yaml_from_dict(state.to_dict(), width=width)
```

Comment:
Actually, it turns out to have additional stuff like 'unresolved_answers'. I realize I left that bit in the controller by accident -- and then I thought, we really shouldn't special-case this, and did this instead. I plan to remove the bit in the controller, but what do you think?